In [1]:
import numpy as np

In [2]:
def lineFit(x, y, s):
    N=x.size
    xgorr=(x/s**2).sum()/((1/s**2).sum())
    ygorr=(y/s**2).sum()/((1/s**2).sum())
    b = ((y * (x - xgorr))/s**2).sum()/ (((x * (x - xgorr))/s**2).sum())
    a = ygorr - b * xgorr
    f=b*x+a
    Chi_cuad=(((y-f)/s)**2).sum()
    Chi_cuad_red=Chi_cuad/(N-2)
    sigma_b_cuad=1/(((x-xgorr)*x/s**2).sum())
    sigma_a_cuad=sigma_b_cuad*(((x**2/s**2).sum())/((1/s**2).sum()))
    return b, a, Chi_cuad_red, sigma_b_cuad, sigma_a_cuad

In [3]:
t,v,s = np.loadtxt("VelocityVsTimeData.txt",skiprows = 3,usecols = (0,1,2),unpack = True)

In [97]:
Slope , V0 , Chi_cuad_red, sigma_b_cuad, sigma_a_cuad = lineFit(t,v,s)
print(Slope , V0 , Chi_cuad_red, sigma_b_cuad, sigma_a_cuad)
print(abs((Slope-(-9.81))/(-9.81))*100)

-10.09217678546556 185.48457346811045 0.7255301700054672 0.274076178974438 38.02092300813783
2.876419831453201


In [98]:
def lineFit(x, y):
    xavg = x.mean()
    slope = (y * (x - xavg)).sum() / (x * (x - xavg)).sum()
    yint = y.mean() - slope * xavg
    return slope, yint

In [99]:
a, v0 = lineFit(t,v)x|
print(v0,a)
print(abs((a-(-9.8))/(-9.8))*100)

171.89273556960842 -9.23414248039144
5.774056322536326


El método de mínimos cuadrados lineal con incertidumbr en los datos resultó ser más exacto pues el error fue de solo el 2.87% cuándo el error de mínimos cuadrados normal fue de 5.77% comparado con el valor exacto.